### PR TITLE
feat: allow filtering of squads

### PR DESCRIPTION
**Pull request by rebelchris**
Added filtering option to set which sourceTypes to return.

Related Frontend: https://github.com/dailydotdev/apps/pull/3612

AS-625 #done

### CODE REVIEW

File: src/common/feedGenerator.ts [hunk 1]
```diff
@@ -355,6 +369,12 @@ export function feedResolver<
       return excludedTypes ? !excludedTypes.includes(type) : true;
     });
 
+    const sourceTypes = baseFeedConfig.source_types.filter(
+      (el) =>
+        !(queryParams as AnonymousFeedFilters).excludeSourceTypes.includes(el),
+    );
+    console.log('qp', sourceTypes);
```
Comment (rebelchris) [inline src/common/feedGenerator.ts:376]:
```suggestion
```

File: src/common/personalizedDigest.ts [hunk 2]
```diff
@@ -216,6 +217,9 @@ export const getPersonalizedDigestEmailPayload = async ({
     blocked_tags: feedConfig.blockedTags,
     blocked_sources: feedConfig.excludeSources,
     feed_config_name: feature.feedConfig,
+    source_types: baseFeedConfig.source_types.filter(
+      (el) => !feedConfig.excludeSourceTypes.includes(el),
+    ),
```
Comment (capJavert) [inline src/common/personalizedDigest.ts:222]: Digest config supports it, I know in the past we had to add/enabled specifically, maybe check with Vas?

File: src/common/feedGenerator.ts [hunk 3]
```diff
@@ -278,6 +286,12 @@ export const applyFeedWhere = (
     );
   }
 
+  if (sourceTypes.length > 0) {
+    newBuilder = newBuilder.andWhere(`source.type IN (:...sourceTypes)`, {
+      sourceTypes,
+    });
+  }
```
Comment (capJavert) [inline src/common/feedGenerator.ts:293]: Index might be needed for this?

File: src/common/feedGenerator.ts [hunk 4]
```diff
@@ -162,23 +163,28 @@ export const feedToFilters = async (
...
-        if (curr.group === 'content_types') {
-          acc.excludeTypes.push(curr.options.type);
-        }
-        if (curr.group === 'content_curation') {
-          acc.blockedContentCuration.push(curr.options.type);
+  const { excludeTypes, blockedContentCuration, excludeSourceTypes } =
+    settings.reduce<{
+      excludeTypes: string[];
+      blockedContentCuration: string[];
+      excludeSourceTypes: string[];
```
Comment (capJavert) [inline src/common/feedGenerator.ts:170]: We need to add this to `FiltersInput` gql type?

**Review by capJavert — COMMENTED** [review]
Some questions mostly

File: src/common/personalizedDigest.ts (hunk 2 above)
Comment (rebelchris) [reply src/common/personalizedDigest.ts:222]: Yeah checked, he's adding it to all:
https://dailydotdev.slack.com/archives/C04UN615UAF/p1727855255223979

File: src/common/feedGenerator.ts (hunk 4 above)
Comment (rebelchris) [reply src/common/feedGenerator.ts:170]: They actually never really come from frontend though, but think we might use it via internal graph resolver on API

File: src/common/feedGenerator.ts (hunk 3 above)
Comment (rebelchris) [reply src/common/feedGenerator.ts:293]: Index already exists.

**Review by capJavert — APPROVED** [review]
Looks alright to me 👍

**Review by omBratteng — APPROVED** [review]
LGTM, just need to be sure it actually gets passed on to feed.
When I added the video filtering, I only had it working for feed v1 🙈

**Pull request merged**